test/libsolidity/syntaxTests/abiEncoder/abi_encodeCall_tuple_from_expression.sol

```diff
@@ -0,0 +1,44 @@
+contract C {
+    event Ev();
+    error Er();
+
+    function g0() internal {}
+    function g1() internal returns (uint) { return (1); }
+    function g2() internal returns (uint, uint) { return (2, 3); }
+
+    function f0() public {}
+    function f1(uint) public {}
+    function f2(uint, uint) public {}
+
+    function h() public view {
+        uint a;
+        uint b;
+
+        abi.encodeCall(this.f0, true ? g0() : g0());
+        abi.encodeCall(this.f1, true ? (1) : (2));
+        abi.encodeCall(this.f1, true ? g1() : g1());
+        abi.encodeCall(this.f2, true ? g2() : g2());
+        abi.encodeCall(this.f2, true ? (1, 2) : (3, 4));
+
+        abi.encodeCall(this.f1, (1) + (2));
+        abi.encodeCall(this.f2, (1, 1) + (2, 2));
+
+        abi.encodeCall(this.f1, (a) = g1());
+        abi.encodeCall(this.f2, (a, b) = g2());
+
+        abi.encodeCall(this.f0, Ev() / Er());
+
+        abi.encodeCall(this.f0, !());
```

Comment:
Here again, I'd add a short comment after each one that's supposed to work. And might be better to just put them in a separate test. Both to make it easier to match the errors to bad ones and because some errors may prevent other errors from showing up so the fact that here a specific line has no error does not 100% it works.

Reply:
Ok. I separated them in more specific files. Also added a few more cases to some tests.